index.js

```diff
@@ -645,7 +645,9 @@ export function load_placements(force_load = false) {
   // elements. If any of these elements have issues, this main promise will
   // reject.
   if (elements.length === 0) {
-    throw new Error("No ad placements found.");
+    // Return a promise that is immediately rejected, instead of a native
+    // exception.
+    return Promise.reject(new Error("No ad placements found."));
```

Comment:
I just merged https://github.com/readthedocs/ethical-ad-client/pull/130 which downgrades this to a simple debug statement.

Reply:
Great, I'll back out this change then :+1:

Reply:
Okay, got a chance to rebase this finally. Should be good now.